cmd/cloud-controller-manager/main.go

```diff
@@ -57,64 +55,51 @@ const (
 func main() {
 	rand.Seed(time.Now().UnixNano())
 
-	pflag.CommandLine.ParseErrorsWhitelist.UnknownFlags = true
-	_ = pflag.CommandLine.Parse(os.Args[1:])
-
-	// this is an example of allow-listing specific controller loops
-	controllerList := []string{"cloud-node", "cloud-node-lifecycle", "service", "route"}
-
-	s, err := options.NewCloudControllerManagerOptions()
+	ccmOptions, err := options.NewCloudControllerManagerOptions()
 	if err != nil {
 		klog.Fatalf("unable to initialize command options: %v", err)
 	}
-	c, err := s.Config(controllerList, app.ControllersDisabledByDefault.List())
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
-	}
 
-	cloud, err := cloudprovider.InitCloudProvider(cloudProviderName, c.ComponentConfig.KubeCloudShared.CloudProvider.CloudConfigFile)
-	if err != nil {
-		klog.Fatalf("Cloud provider could not be initialized: %v", err)
-	}
-	if cloud == nil {
-		klog.Fatalf("cloud provider is nil")
-	}
+	cloudInitializer := func(config *cloudcontrollerconfig.CompletedConfig) cloudprovider.Interface {
```

Comment:
I think the anonymous function makes this harder to read/understand. Can we move this out to its own named function?

Reply:
I was thinking to give cloud provider the power to initialize cloud with the cloud provider name they specified so put it in main.go.

Reply:
No objection to having the code in main.go. My objection is to having a disjoint section of anonymous code in the middle of this function when we're not executing it here or now. If we create a named function with this code in it, we can pass it later where we need it and the code will read more easily.

Reply:
Totally make sense. Let me update. Thank you ^^